pkg/virt-controller/services/template.go

```diff
@@ -1399,13 +1391,7 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, t
 		pod.Spec.PriorityClassName = vmi.Spec.PriorityClassName
 	}
 
-	if vmi.Spec.Affinity != nil {
-		pod.Spec.Affinity = vmi.Spec.Affinity.DeepCopy()
-	}
-
-	if t.clusterConfig.CPUNodeDiscoveryEnabled() {
-		SetNodeAffinityForForbiddenFeaturePolicy(vmi, &pod)
-	}
+	t.updateAffinity(vmi, &pod, attachmentPod)
```

Comment:
This won't work.

The attachment pod needs to be assigned to the VMI pod, not the other way around. The VMI pod needs to be scheduled without taking the attachment pod into account. 

Schedule the VMI pod, then schedule the attachment pod to the VMI pod, then hand off everything to virt-handler.